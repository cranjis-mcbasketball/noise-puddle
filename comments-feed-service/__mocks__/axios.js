'use strict';

const Promise = require('bluebird');

module.exports = {
  get: (path) => {
    switch(path) {
      case 'http://localhost:3005/artist':
        return Promise.resolve({
          data: [
            {
              "id": 1,
              "name": "testArtistName",
              "followers_count": 999999,
              "tracks_count": 999,
              "profile_pic": "",
              "createdAt": "2020-03-18T15:36:29.000Z",
              "updatedAt": "2020-03-18T15:36:29.000Z"
            }
          ]
        })
        break;

      case 'http://localhost:3005/song':
        return Promise.resolve({
          data: [
            {
              "id": 1,
              "artist_id": 1,
              "title": "Test Song Title",
              "play_count": 999999,
              "likes_count": 999,
              "repost_count": 99,
              "description": "",
              "released_by": "",
              "release_date": "01 January 2020",
              "p_line": "â„— 2020",
              "c_line": "Â© 2019",
              "hashtags": "",
              "createdAt": "2020-03-18T15:36:29.000Z",
              "updatedAt": "2020-03-18T15:36:29.000Z"
            }
          ]
        })
        break;

      case 'http://localhost:3005/comments':
        return Promise.resolve({
          data: [
            {
              "id": 998,
              "user_id": "d2cc14dd-e8ad-44d2-a8f0-86262928test",
              "user_name": "Test User 1",
              "user_followers_count": 5,
              "user_profile_pic": "https://fec-comments-images.s3.us-east-2.amazonaws.com/zelda.jpg",
              "text": "Learning how to write useful tests is hard!",
              "track_location": "01:00",
              "original_comment_id": null,
              "createdAt": "2020-03-23T01:48:48.000Z",
              "updatedAt": "2020-03-23T01:48:48.000Z"
            },
            {
              "id": 999,
              "user_id": "14914f57-b595-4196-92ea-e2b6bf1ftest",
              "user_name": "Test User 2",
              "user_followers_count": 99,
              "user_profile_pic": "https://fec-comments-images.s3.us-east-2.amazonaws.com/zelda.jpg",
              "text": "One day writing tests will be easier!",
              "track_location": "03:00",
              "original_comment_id": null,
              "createdAt": "2020-03-23T01:47:52.000Z",
              "updatedAt": "2020-03-23T01:47:52.000Z"
            }
          ]
        })
        break;
    }
  },
  post: () => {
    return Promise.resolve({
      data: [
        'OK'
      ]
    })
  }
};