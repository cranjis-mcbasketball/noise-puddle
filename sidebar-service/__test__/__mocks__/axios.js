'use strict';
module.exports = {
  get: () => {
    return Promise.resolve({
      data: {
        "artist": {
          "id": 15,
          "name": "Pettifogger_Gastromancy",
          "track_count": 133,
          "follower_count": 7386,
          "following_count": 627,
          "about": "Put a bird on it readymade locavore beard tattooed XOXO normcore yuccie synth taiyaki brooklyn copper mug. Typewriter put a bird on it 8-bit four dollar toast. Celiac pork belly pug street art knausgaard ramps shaman tote bag put a bird on it church-key. Literally food truck vexillologist hot chicken chambray artisan. Slow-carb vice taxidermy, ethical vape squid la croix photo booth celiac shoreditch af pour-over dreamcatcher. ",
          "links": "https://clarissa.org http://rafael.com http://pauline.biz",
          "liked_songs": 5
        },
        "likedSongs": [
          {
            "id": 1,
            "song_name": "Cone BBQ",
            "artist_name": "Tatterdemalion_Lollygag",
            "plays": 86897370,
            "likes": 1903668,
            "reposts": 32408,
            "comments": 9357,
            "album_art": "https://fec-sidebar-album-art.s3.amazonaws.com/abba/Mamma_Mia_Intermezzo_No_1.jpg",
            "location": "New Lysanneberg, Cuba",
            "artist_pic": "https://fec-sidebar-album-art.s3.amazonaws.com/abba/Mamma_Mia_Intermezzo_No_1.jpg"
          },
          {
            "id": 3,
            "song_name": "BBQ Breakfast",
            "artist_name": "Collywobbles_Maverick",
            "plays": 68902784,
            "likes": 4006950,
            "reposts": 146820,
            "comments": 33039,
            "album_art": "https://fec-sidebar-album-art.s3.amazonaws.com/abba/R-3365228-1505192441-2722.jpeg.jpg",
            "location": "Port Bonita, Tokelau",
            "artist_pic": "https://fec-sidebar-album-art.s3.amazonaws.com/abba/R-3365228-1505192441-2722.jpeg.jpg"
          },
          {
            "id": 5,
            "song_name": "Boat Dog",
            "artist_name": "Quean_Lickspittle",
            "plays": 50762992,
            "likes": 1051195,
            "reposts": 20048,
            "comments": 5264,
            "album_art": "https://fec-sidebar-album-art.s3.amazonaws.com/abba/ABBA-Album-Covers-6.jpg",
            "location": "Isidrotown, Indonesia",
            "artist_pic": "https://fec-sidebar-album-art.s3.amazonaws.com/abba/ABBA-Album-Covers-6.jpg"
          }
        ]
      }
    });
  }
};