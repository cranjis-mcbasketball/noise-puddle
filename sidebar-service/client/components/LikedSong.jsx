import React from 'react';

import styled from 'styled-components';

import { TiHeart, TiArrowLoop } from 'react-icons/ti';
import { MdPlayArrow } from 'react-icons/md';
import { IoIosText} from 'react-icons/io';

const Li = styled.li`
  padding: 5px 0;
  display: list-item;
  list-style: none;
  margin: 0;
`;

const SongDiv = styled.div`
  padding: 0;
  left: -6px;
  margin-right: -6px;
  position: relative;
  min-height: 60px;
  border-radius: 4px;
  overflow: hidden;
  `;

const AlbumArtDiv = styled.span`
  height: 50px;
  width: 50px;
  text-align: center;
  position: relative;
  background-repeat: no-repeat;
`;

const AlbumArtSpan = styled.span`

  margin-right: 4px;
  padding: 5px 6px;
  width: 50px;
  height: 50px;
  position: relative;
  float: left;
`;

const MediaInfoDiv = styled.span`
  margin-top: 4px;
  overflow: hidden;
  display: block;
`;

const AlbumImage = styled.img`
  width: 50px;
  height: 50px;
  opacity: 1;
`;

const SongArtist = styled.div`
  margin-top: 0;
  fontSize: 14px;
  font-family: Interstate,Lucida Grande,Lucida Sans Unicode,Lucida Sans,Garuda,Verdana,Tahoma,sans-serif;
  font-weight: 400;
  word-wrap: break-word;

  flex: 1;
  min-width: 0;
}
`;

const ArtistDiv = styled.div`
overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
    word-break: normal;
        font-weight: 100;
    color: #999;
    display: block;
    list-style: none;
line-height: 1;
    margin: 0 0 2px;
`;

const ArtistNameSpan = styled.span`
  font-weight: 100;
  line-height: 1;
  margin-bottom: 2px;
  cursor: pointer;
`;

const SongNameDiv = styled.div`
  height: 1.3em;

  text-decoration: none;
  color: #333;
  display:block;
  max-width: 100%;
  font-weight: 100;
  box-sizing: border-box;
  white-space: nowrap;
  text-overflow: ellipsis;
  word-break: normal;
`;

const SmallUl = styled.ul`

    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;
    list-style: none;
    margin: 0px;
    padding: 0px;
    margin-block-start: 0px;
    margin-block-end: 0px;
    margin-inline-start: 0px;
    margin-inline-end: 0px;


`;

const SmallLi = styled.li`
  display: list-item;
  text-align: -webkit-match-parent;
  list-style: none;
`;

const SmallSpan = styled.span`
  margin-right: 9px;
  margin-bottom: 5px;
  color: #999;
  fontSize: 11px;
  line-height: 12px;
  display: inline-block;
  font-family: Interstate,Lucida Grande,Lucida Sans Unicode,Lucida Sans,Garuda,Verdana,Tahoma,sans-serif;
  font-weight: 100;
  font-feature-settings: "tnum";
  font-variant-numeric: tabular-nums;

  display: inline-block;
  content: "";
  vertical-align: top;
  background-repeat: no-repeat;
  background-position: 50%;
`;



const LikedSong = ({song_name, artist_name, plays, likes, reposts, comments, album_art, location, artist_pic, onArtistNameClick, numberConversion}) => (
  <Li className='INDIVIDUALLIKEDSONGS'>
    <SongDiv className='CONTENTSOFLIKEDSONG'>
      <AlbumArtDiv>
        <AlbumArtSpan className='ALBUMART'>
          <AlbumImage src={album_art}></AlbumImage>
        </AlbumArtSpan>
      </AlbumArtDiv>

      <MediaInfoDiv className='MEDIA'>
        <SongArtist className='SONGNAMEARTISTINFO'>
          <ArtistDiv className='ARTISTNAME'>
            <ArtistNameSpan onClick={onArtistNameClick}>{artist_name}</ArtistNameSpan>
          </ArtistDiv>
          <SongNameDiv className='SONGNAME'>
            <span>{song_name}</span>
          </SongNameDiv>

        </SongArtist>

        <SmallUl className='MORESTATS'>
          <SmallLi><SmallSpan><MdPlayArrow />{numberConversion(plays)}</SmallSpan></SmallLi>
          <SmallLi><SmallSpan><TiHeart /> {numberConversion(likes)} </SmallSpan></SmallLi>
          <SmallLi><SmallSpan><TiArrowLoop /> {numberConversion(reposts)} </SmallSpan></SmallLi>
          <SmallLi><SmallSpan><IoIosText/> {numberConversion(comments)} </SmallSpan></SmallLi>
        </SmallUl>
      </MediaInfoDiv>
    </SongDiv>
  </Li>

);

export default LikedSong;