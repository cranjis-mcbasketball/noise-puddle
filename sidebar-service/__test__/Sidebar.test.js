//const app = require('../app.js');
import React from 'react';
import Sidebar from '../client/components/Sidebar.jsx';
import Links from '../client/components/Links.jsx';
import LikedSong from '../client/components/LikedSong';
import renderer from 'react-test-renderer';

import axios from 'axios';
import {configure, shallow, mount, render} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

jest.mock('axios');

configure({adapter: new Adapter()});


// Test add new artist works
test('Four is not divisible by three', () => {
  expect( 4 % 3 !== 0).toBe(true);
});

// Test get requests from App
describe('Sidebar component', () => {
  describe('When rendered', () => {
    it('should fetch an object with artist and liked songs', () => {

      const spyMount = jest.spyOn(axios, 'get');
      const SidebarInstance = shallow(
        <Sidebar/>
      );

      expect(spyMount).toBeCalled();
    });
  });
});

// Test get requests from LikedSong
describe('Sidebar component', () => {
  describe('When clicked', () => {
    it('should run get request for an object with artist and liked songs', () => {
      const testfunc = function() {console.log('hey')}

      const spyMount = jest.spyOn(axios, 'get');
      const LikedSongWrapper = shallow(<LikedSong onClick={spyMount} e={{ currentTarget: { textContent: 'Pettifogger_Gastromancy' }}} numberConversion={testfunc}/>);

      LikedSongWrapper.find('.ARTISTNAME').simulate('click');
      expect(spyMount).toBeCalled();
    });
  });
});

// Snapshot test for sidebar rendering.
test('Components render on mount', () => {
  const sidebar = renderer.create(<Sidebar />);
  var component = sidebar.toJSON();
  expect(sidebar).toMatchSnapshot();
});