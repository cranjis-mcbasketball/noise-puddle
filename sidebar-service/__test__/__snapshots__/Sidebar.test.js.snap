// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components render on mount 1`] = `
<div
  className="Sidebar__SidebarContainer-sc-1qcjuen-0 gXyDFV"
  id="sidebar-right"
>
  <article
    id="stats"
  >
    <table
      className="Sidebar__StatTable-sc-1qcjuen-1 bTgztn"
      id="links"
    >
      <tbody
        className="Sidebar__Tbody-sc-1qcjuen-2 gEreah"
      >
        <tr
          className="Sidebar__Tr-sc-1qcjuen-3 csmPta"
        >
          <td
            className="StatCounts__Td-dpcx90-0 eAZwhG stat-column"
          >
            <div
              className="StatCounts__IndentedDiv-dpcx90-1 ftFxrc"
            >
              <h3
                class="stat-title"
                className="StatCounts__H3-dpcx90-2 dphLZd"
              >
                followers
              </h3>
              <div
                class="stat-num"
                className="StatCounts__StatNumDiv-dpcx90-3 gRKrvn"
              />
            </div>
          </td>
          <td
            className="StatCounts__Td-dpcx90-0 eAZwhG stat-column"
          >
            <div
              className="StatCounts__IndentedDiv-dpcx90-1 ftFxrc"
            >
              <h3
                class="stat-title"
                className="StatCounts__H3-dpcx90-2 dphLZd"
              >
                following
              </h3>
              <div
                class="stat-num"
                className="StatCounts__StatNumDiv-dpcx90-3 gRKrvn"
              />
            </div>
          </td>
          <td
            className="StatCounts__Td-dpcx90-0 eAZwhG stat-column"
          >
            <div
              className="StatCounts__IndentedDiv-dpcx90-1 ftFxrc"
            >
              <h3
                class="stat-title"
                className="StatCounts__H3-dpcx90-2 dphLZd"
              >
                tracks
              </h3>
              <div
                class="stat-num"
                className="StatCounts__StatNumDiv-dpcx90-3 gRKrvn"
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </article>
  <article>
    <div
      className="Sidebar__AboutDiv-sc-1qcjuen-4 jJTYND"
      id="about"
    >
      <div>
        <div>
          <div
            style={
              Object {
                "position": "absolute",
                "z-index": "-1",
              }
            }
          >
            <div
              className="About__Invis-sc-9ahlt-3 eYwoRx"
            >
              <div
                className="About__AboutDiv2Big-sc-9ahlt-1 dmoGpK"
                id="notCollapsed"
              >
                <div
                  id="wrapper"
                >
                  <div
                    id="truncated-description"
                  >
                    
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="About__AboutDiv2-sc-9ahlt-0 kKpAKn"
            id="notCollapsed"
          >
            <div
              id="wrapper"
            >
              <div
                id="truncated-description"
              >
                
              </div>
            </div>
          </div>
        </div>
        <div>
          <div />
        </div>
      </div>
    </div>
  </article>
  <article>
    <div
      className="Sidebar__LinksDiv-sc-1qcjuen-5 iBUgtV"
      id="weblinks"
    >
      <ul
        className="Links__Ul-sc-2fy7xk-0 bRdtPc"
      />
    </div>
  </article>
  <div
    className="Sidebar__ArticleLiked-sc-1qcjuen-6 jxzEiT"
  >
    <div
      id="likedsongs"
    >
      <div>
        <div
          className="LikedSongs__MoreStyling-y57efl-0 bVTIEM"
        >
          <h3
            className="LikedSongs__H3-y57efl-1 jaekZh LIKEDSONGCLASS"
          >
            <span
              className="LIKEDSONGSNUMBER"
            >
              <svg
                baseProfile="tiny"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                version="1.2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 10.375c0-2.416-1.959-4.375-4.375-4.375s-4.375 1.959-4.375 4.375c0 1.127.159 2.784 1.75 4.375l7 5.25s5.409-3.659 7-5.25 1.75-3.248 1.75-4.375c0-2.416-1.959-4.375-4.375-4.375s-4.375 1.959-4.375 4.375"
                />
              </svg>
               
               likes
            </span>
          </h3>
        </div>
        <span
          className="LikedSongs__SongListDiv-y57efl-4 eLTmdf"
        >
          <span
            className="LikedSongs__Ul-y57efl-5 crPpxo"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
